scripts/build_fuzzers.py

```diff
@@ -9,44 +11,63 @@
 G_BUCKET = 'gs://fuzzer_targets/master/'
 ARCH_CONFIG_NAME = 'x86_64-unknown-linux-gnu'
 BUILDER_START_TIME = time.strftime('%Y%m%d%H%M%S', time.gmtime())
+TAR_NAME = f'fuzz-targets-{BUILDER_START_TIME}.tar.gz'
 
 
-def push_to_google_bucket(runner: str) -> None:
-    tar_name = f'{runner}-master-{BUILDER_START_TIME}.tar.gz'
-    with tarfile.open(name=tar_name, mode='w') as archiver:
-        archiver.add(f"target/{ARCH_CONFIG_NAME}/release/{runner}", runner)
+def push_to_google_bucket(archive_name: str) -> None:
+    try:
+        subprocess.run([
+            'gsutil',
+            'cp',
+            archive_name,
+            G_BUCKET,
+        ], check=True)
 
-    _proc = subprocess.run([
-        'gsutil',
-        'cp',
-        tar_name,
-        G_BUCKET,
-    ], check=True)
+    except subprocess.CalledProcessError as e:
+        logger.info(f"Failed to upload archive to Google Storage!")
 
 
 def main() -> None:
+    fuzz_bin_list = []
+
     crates = [
         i for i in toml.load('nightly/fuzz.toml')['target']
+        # temporary limit for fuzz targets building
         if i['runner'] in ['borsh', 'serde']
     ]
     logger.debug(f"Crates from nightly/fuzz.toml: \n{crates}")
 
     for target in crates:
         logger.info(f"working on {target}")
+        runner = target['runner']
+        try:
+            subprocess.run(
+                [
+                    'cargo',
+                    '+nightly',
+                    'fuzz',
+                    'build',
+                    runner,
+                ],
+                check=True,
+                cwd=target['crate'],
+            )
+            fuzz_bin_list.append(f"target/{ARCH_CONFIG_NAME}/release/{runner}")
+
+        except subprocess.CalledProcessError as e:
+            logger.info(f"Failed to build/archive target: {target}")
+
+    logger.info(
+        f"Fuzzer binaries to upload: {[os.path.basename(f) for f in fuzz_bin_list]}"
+    )
+
+    # adding to archive in 1 operation as tarfile doesn't
+    # support appending to existing compresed file
+    with tarfile.open(name=TAR_NAME, mode="w:gz") as archive:
+        for file_name in fuzz_bin_list:
+            archive.add(file_name, os.path.basename(file_name))
```

Comment:
Hmm do you think it would work without taking the basename here? My guess would be probably but we need to try; I’ll upload a new archive with the fuzzers not-at-top-level and see if that works out fine :)

Reply:
To expand on this: right now we have `core/account-id/fuzz borsh` and `core/account-id/fuzz serde` fuzzers that actually fuzz account-id in two different ways. With time we’ll probably have other fuzzers, say `core/primitives/fuzz borsh` that’d also test borsh (de)serialization of things in the `primitives` crate, at which point we won’t be able to have both `borsh`-named targets at the top-level.

Reply:
It should work because it does a walk through all extracted path tree https://github.com/google/clusterfuzz/blob/master/src/clusterfuzz/_internal/bot/fuzzers/utils.py#L91